### ISSUE TITLE
Use serialization library for storing / loading state of pipeline modules?

### ISSUE THREAD

**Issue by damellis** [nickgillian/grt#66]
It might be nice to use something like [Cereal](http://uscilab.github.io/cereal/) or [boost::serialization](http://www.boost.org/doc/libs/1_61_0/libs/serialization/doc/) for saving and loading the state of pre-processing, feature extraction, classification modules, etc. That would simplify the process of writing new modules by reducing the amount of boilerplate saving and loading code required.

**Comment by nickgillian** [comment]
Yes, this would be great, particularly for reducing the boilerplate code required for writing new modules.

The key thing here is how to add this sort of support without bringing in a major dependency like boost (a great library, but I'm trying to keep the toolkit free from major dependencies like this).

**Comment by damellis** [comment]
Good point. [Cereal](http://uscilab.github.io/cereal/quickstart.html) seems relatively light-weight: it looks like its headers-only and doesn't require any dependencies. It's still an external code-base, but it's a relatively small one.

**Comment by cyberluke** [comment]
I'm now doing this manually - simple txt file with parameters to save it together with training data.
